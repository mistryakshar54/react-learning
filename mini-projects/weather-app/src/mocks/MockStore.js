export default class MockStore {
  constructor(initialState = {}) {
    this.dispatch = jest.fn();
    this.subscribe = jest.fn();
    this.getState = jest.fn().mockReturnValue(initialState);
    this.replaceReducer = jest.fn();
  }
}
