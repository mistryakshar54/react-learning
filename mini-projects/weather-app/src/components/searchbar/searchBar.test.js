import React from 'react';
import { mount } from 'enzyme';
import SearchBarComponent from './searchBar';
import { Provider } from 'react-redux';
import MockStore from '../../mocks/MockStore';
test('it shallow mounts the component' , () => {
    const store = new MockStore();
    const dispatch = jest.fn();
    const handleLocSearch = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <SearchBarComponent />
      </Provider>
    );
    const input = wrapper.find("input");
    // console.log("Input" , input.val())
    input.simulate("change", { target: { value: "Hello" } });
    // console.log("Input Changed" , input.val())
    expect(handleLocSearch).toHaveBeenCalled();

    expect( wrapper).toMatchSnapshot();
});