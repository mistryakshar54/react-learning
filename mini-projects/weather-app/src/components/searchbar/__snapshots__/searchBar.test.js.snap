// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it shallow mounts the component 1`] = `ReactWrapper {}`;
